forc-plugins/forc-debug/src/cli/mod.rs

```diff
@@ -0,0 +1,155 @@
+                            }
+                        }
+                        cmd if helper.commands.is_register_command(cmd) => {
+                            if let Err(e) = commands::cmd_registers(&mut state, args).await {
+                                println!("Error: {}", e);
+                            }
+                        }
+                        cmd if helper.commands.is_breakpoint_command(cmd) => {
+                            if let Err(e) = commands::cmd_breakpoint(&mut state, args).await {
+                                println!("Error: {}", e);
+                            }
+                        }
+                        cmd if helper.commands.is_memory_command(cmd) => {
+                            if let Err(e) = commands::cmd_memory(&mut state, args).await {
+                                println!("Error: {}", e);
+                            }
+                        }
+                        cmd if helper.commands.is_quit_command(cmd) => {
+                            break;
+                        }
+                        cmd if helper.commands.is_reset_command(cmd) => {
+                            if let Err(e) = commands::cmd_reset(&mut state, args).await {
+                                println!("Error: {}", e);
+                            }
+                        }
+                        cmd if helper.commands.is_continue_command(cmd) => {
+                            if let Err(e) = commands::cmd_continue(&mut state, args).await {
+                                println!("Error: {}", e);
+                            }
+                        }
+                        cmd if helper.commands.is_step_command(cmd) => {
+                            if let Err(e) = commands::cmd_step(&mut state, args).await {
+                                println!("Error: {}", e);
+                            }
+                        }
+                        unknown_cmd => {
+                            if let Some(suggestion) = helper.commands.find_closest(unknown_cmd) {
+                                println!(
+                                    "Unknown command: '{}'. Did you mean '{}'?",
+                                    unknown_cmd, suggestion.name
+                                );
+                            } else {
+                                println!("Unknown command: '{}'", unknown_cmd);
+                            }
+                        }
+                    }
+                }
+            }
+            Err(rustyline::error::ReadlineError::Interrupted) => {
+                println!("CTRL-C");
+                break;
+            }
+            Err(rustyline::error::ReadlineError::Eof) => {
+                println!("CTRL-D");
+                break;
+            }
+            Err(err) => {
+                println!("Error: {}", err);
+                break;
+            }
+        }
+    }
+
+    // Save history
+    let _ = editor.save_history(&history_path);
```

Comment:
Maybe it would be worth putting `save_history()` in `Drop` so that we save even if the user does a `ctrl-C`?

Reply:
good point!

Reply:
yeah good idea, just implementing this now.

Reply:
done [a514c84](https://github.com/FuelLabs/sway/pull/6830/commits/a514c84c2cd89e6c89599b892f5dca9fd9d3badc)